dashboard/app.py

```diff
@@ -68,22 +68,42 @@ def main():
 
 @app.route('/leaderboard')
 def leaderboard():
-    page_content = cache.get('leaderboard')
+    page = int(request.args.get('page', 1))
+    if page < 1:
+        abort(400)
+
+    page_content = cache.get('leaderboard-page-{}'.format(page))
     if page_content is None:
+        offset = (page - 1) * LEADERBOARD_ROWS_PER_PAGE
         date_from, date_to = get_week_range(datetime.utcnow().date())
-        leader_board_rows = get_leaderboard_rows(date_from, date_to, 10)
-        enrich_leaderboard_rows(leader_board_rows, date_from, date_to)
+        leaderboard_rows = get_leaderboard_rows(
+            date_from, date_to, offset=offset, limit=LEADERBOARD_ROWS_PER_PAGE
+        )
+        enrich_leaderboard_rows(leaderboard_rows, date_from, date_to)
+
+        previous_page = None
+        if page >= 2:
+            previous_page = page - 1
+
+        next_page = None
+        if len(leaderboard_rows) == LEADERBOARD_ROWS_PER_PAGE:
```

Comment:
This is not always correct - i.e. if we have 10 rows and we're in first page, next button should be disabled.

Reply:
This is a drawback of solution where total rows count is not known. Any ideas how to improve this? :)

Reply:
Ok, this works for now. Can we add a TODO here to remember this issue?

Reply:
`+`